packages/gestalt/src/Button.js

```diff
@@ -142,7 +157,89 @@ const ButtonWithForwardRef: React$AbstractComponent<
     </Text>
   );
 
+  const iconEndComponent = (): Node => {
+    const SIZE_NAME_TO_PIXEL = {
+      sm: 10,
+      md: 12,
+      lg: 12,
+    };
+
+    return (
+      <Box alignItems="center" display="flex">
+        {buttonText}
+        <Box display="inlineBlock" flex="none" marginStart={2}>
+          <Icon
+            accessibilityLabel=""
+            color={textColor}
+            icon={iconEnd}
+            size={SIZE_NAME_TO_PIXEL[size]}
+          />
+        </Box>
+      </Box>
+    );
+  };
+
   /* eslint-disable react/button-has-type */
+  if (props.type === 'link') {
+    const { href, rel, target } = props;
+
+    return (
+      <Link
+        hoverStyle="none"
+        href={href}
+        onClick={({ event }) => {
+          if (onClick) {
+            onClick({ event });
+          }
+        }}
+        ref={innerRef}
+        rel={rel}
+        target={target}
+        inline={inline}
+        rounding="pill"
+        tapStyle="compress"
```

Comment:
@AlbertCarreras doesn't this mean we have 2 compress styles?
1. Within `classes` on the wrapping `div`
2. Within `Link`

If so, we probably want to keep the one on `Link` and make the one on `classes` conditional.

```
[touchableStyles.tapCompress]: props.type !== 'link' && !disabled && isTapping
```